### PR TITLE
4958969: ObjectOutputStream example leads to non-working code

### PR DESCRIPTION
**Pull request by RogerRiggs**
Update javadoc examples to use try-with-resources and use javadoc snippet tags.
The examples in ObjectInputStream and ObjectOutputStream get an update with
try-with-resources and use of javadoc snippet tags.
ObjectInputFilter is updated with use of snippet tags instead of html markup.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-4958969](https://bugs.openjdk.org/browse/JDK-4958969): ObjectOutputStream example leads to non-working code

### Reviewers
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**) ⚠️ Review applies to [c27b0e5a](https://git.openjdk.org/jdk20/pull/59/files/c27b0e5a2c85722457443d2e311254cecb95ca85)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk20 pull/59/head:pull/59` \
`$ git checkout pull/59`

Update a local copy of the PR: \
`$ git checkout pull/59` \
`$ git pull https://git.openjdk.org/jdk20 pull/59/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 59`

View PR using the GUI difftool: \
`$ git pr show -t 59`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk20/pull/59.diff">https://git.openjdk.org/jdk20/pull/59.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/io/ObjectInputStream.java [hunk 1]
```diff
@@ -116,18 +116,18 @@
...
- *      Date date = (Date) ois.readObject();
- *
- *      ois.close();
- * </pre>
+ * {@snippet lang="java" :
+ *     try (FileInputStream fis = new FileInputStream("t.tmp");
+ *          ObjectInputStream ois = new ObjectInputStream(fis)) {
+ *         String label = (String) ois.readObject();
+ *         LocalDateTime date = (LocalDateTime) ois.readObject();
+ *         System.out.println(label + ": " + date);
```
Comment (LanceAndersen) [inline src/java.base/share/classes/java/io/ObjectInputStream.java]: would it be worth using `System.out.printf` here?

**Review by LanceAndersen — APPROVED** [review]
Looks good Roger,  one trivial suggestion to consider below

File: src/java.base/share/classes/java/io/ObjectInputStream.java (hunk 1 above)
Comment (RogerRiggs) [reply src/java.base/share/classes/java/io/ObjectInputStream.java]: Maybe printing is a distraction; there many preferences for output and the original did not include it.
Simple string concatenation is the most efficient, but the least parameterizable.

File: src/java.base/share/classes/java/io/ObjectInputStream.java [hunk 2]
```diff
@@ -116,18 +116,18 @@
...
- *      String today = (String) ois.readObject();
- *      Date date = (Date) ois.readObject();
- *
- *      ois.close();
- * </pre>
+ * {@snippet lang="java" :
+ *     try (FileInputStream fis = new FileInputStream("t.tmp");
+ *          ObjectInputStream ois = new ObjectInputStream(fis)) {
+ *         String label = (String) ois.readObject();
+ *         LocalDateTime date = (LocalDateTime) ois.readObject();
```
Comment (naotoj) [inline src/java.base/share/classes/java/io/ObjectInputStream.java]: Nit: I'd replace `date` with `dateTime`.

**Review by naotoj — APPROVED** [review]
LGTM, Roger.

**Review by naotoj — APPROVED** [review]

**Pull request closed without merging**